src/wasi/WASI.cpp

```diff
@@ -120,77 +175,73 @@ void WASI::fd_seek(ExecutionState& state, Value* argv, Value* result, Instance*
     uint32_t fd = argv[0].asI32();
     uint64_t fileDelta = argv[1].asI32();
     uint32_t whence = argv[2].asI32();
-    uint64_t newOffset = argv[3].asI32();
-
-    uvwasi_filesize_t out_addr = *(instance->memory(0)->buffer() + newOffset);
+    uvwasi_filesize_t* file_size = reinterpret_cast<uvwasi_filesize_t*>(get_memory_pointer(instance, argv[3], sizeof(uvwasi_filesize_t)));
```

Comment:
What if `get_memory_pointer` returns a nullptr here?
In WASI functions, when accessing memory, is range always correct?